x-pack/plugins/security_solution/public/detection_engine/rule_exceptions/utils/helpers.tsx

```diff
@@ -870,3 +886,111 @@ export const enrichSharedExceptions = (
     });
   });
 };
+
+/**
+ * Creates new Rule exception item with passed in entries
```

Comment:
Thanks @vitaliidm for finding this issue 👏🏻 , please find the fix here [78aedc3](https://github.com/elastic/kibana/pull/159029/commits/78aedc38fc04643cb44ea6c985e3ea7b0b5f989e)

Reply:
If we choose to implement this temporary solution using const [currentExceptionListItems, setCurrentExceptionListItems], the following behavior can be expected:

 

https://github.com/elastic/kibana/assets/12671903/bb4dfaf6-df41-4f18-a746-5507b148b16c

Reply:
> It worked perfectly for the Rule Exception, however, I found that we already have the same behavior happening for the Endpoint Exception so I revert back to check if this is a known bug

I would vote to include fix in this PR, since adding exception would be working only in 50% of cases. i.e. won't be populated when open exception window from alert flyout. I don't know which is more common, but personally I always was using action menu from alerts flyout

Reply:
> To address this, I added a temporary state variable const [currentExceptionListItems, setCurrentExceptionListItems] and used it within a useEffect hook:

```ts

  const {
    addNested,
    andLogicIncluded,
    disableAnd,
    disableNested,
    disableOr,
    warningExists,
    exceptions,
    exceptionsToDelete,
  } = state;
```

can `exceptions` from state be used instead of `currentExceptionListItems` ?

Reply:
I agree with you, but I have some concerns about changing this part because it affects the test cases for all Exceptions. It's difficult to anticipate all the specific cases that need to be validated to ensure that they continue to behave as expected. However, I want to note that Cypress has detected some failure scenarios.

On another note, it seems that none of the users have complained about the Endpoint Exception not being populated from the Alert's Summary Flyout action menu. So, I'm uncertain whether we should address this issue in the current pull request or create a new ticket to handle it properly. 🤔 

I will give it a try and see if Cypress will be pass if I added the extra state

Reply:
> can exceptions from state be used instead of currentExceptionListItems ?

Not really, you need a separate state to just block the infinite state change, and if you replaced the `currentExceptionListItems`  with `exceptions` it will block the `Edit` workflow  and keep the Conditions as they are

Reply:
This issue will be handled in another PR, however this is the [bug](https://github.com/elastic/kibana/issues/159784)